Blobstore serve

zghera opened the pull request:
Add functionality to serve the image uploaded to Blobstore (see #20) to the comments thread on the server-dev page.

The integration of this functionality involved the following steps:
- Creation of a new class, CommentsThread that contains two parallel lists. One for the comments' text and one for the imageURLs if they include them (null otherwise).
- Modifications in ListCommentsServlet.java to create an instance of CommentsThread on each Get request and add convert this object to JSON to be published on the comment-data page.
- Modify server-dev.js to create and include an image tag in each comment if there was an image uploaded for that comment.

Testing was done by providing all combinations of inputs to the comment entry form and verifying the actual behavior matched the expected. These included:
- Comment with text and a normal image (png, jpg, etc.). Expected behavior: image and text are included within the comment box for the comment thread.
- Comment with only text (no image upload). Expected behavior: Just text shows up and no image (like before using blobstore).
- Comment with no text but an image. Expected behavior: Only the image shows up.
- Comment with no text or image. Expected behavior: empty comment.
- Comment with text and a file uploaded that is not an image. Expected behavior, only the text shows up and the image url stored in the database is null.

zghera committed:
Initial implementation to serve the text and image for each comment to the comments thread UI.

zghera committed:
Move process to determine number of commentes displayed into own function and improved variable names.

zghera committed:
Fix missing semi-colon.

zghera committed:
Add testing print statements.

zghera committed:
Fix NaN and var/item naming bugs.

First bug fix was comparing a variable to NaN using '==' instead of isNan(). There was also a slew of bugs due to the naming change for the comments thread (json data) and commments thread containers (ul in server-dev.html).

zghera committed:
Force consistent name usage for the comments thread.

zghera committed:
Fix incorrect CommentsThread import.

zghera committed:
Remove debugging print statements.

beiebi0 commented on portfolio/src/main/java/com/google/sps/data/CommentsThread.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class CommentsThread {
+
+  /** List of the strings of text in each comment */
+  private final List<String> texts = new ArrayList<>();
+
+  /** List of the image urls each comment (null if no image for a comment) */
+  private final List<String> imageUrls = new ArrayList<>();
+
```
Based on my understanding, it's better to have a new data structure like 
class Comment {
  String text;
  String imageUrl;
}

then you have List\<Comment\> in this CommentsThread.

zghera replied:
I like this a lot more as well. Done.

zghera committed:
Merge master into branch blobstore-serve.

zghera committed:
Merge branch 'master' into blobstore-serve

zghera committed:
Modify comment thread creation so it is List of Comment objects.

Rather than having a single object, CommentsThread, that contains parallel lists for text and image urls, have a list of objects where each object contains just the text and imageurl for that comment.

zghera committed:
Merge branch 'blobstore-serve' of https://github.com/zghera/STEP-Portfolio into blobstore-serve

zghera committed:
Merge branch 'master' into blobstore-serve
